src/main/java/org/ecocompass/core/K_DTree/KdNode.java

```diff
@@ -1,17 +1,52 @@
 package org.ecocompass.core.K_DTree;
 
+import lombok.Getter;
+import lombok.Setter;
 import org.ecocompass.core.graph.Node;
 
 // Node class representing a point in the k-dimensional space
+@Getter
+@Setter
 public class KdNode {
-    double[] coordinates; // Array of coordinates for each dimension
+    double[] coordinates;
+    Long nodeID;
     KdNode left;
     KdNode right;
 
     Node node;
 
-    public KdNode(double[] coordinates, Node node) {
+    public KdNode(double[] coordinates, Long nodeID, Node node) {
         this.coordinates = coordinates;
         this.node = node;
+        this.nodeID = nodeID;
     }
+
```

Comment:
I know after @Getter @Setter annotations these explicit get public functions are not to be defined. But it was not working and giving me errors so will figure it out later and refactor later.